src/client.rs

```diff
@@ -225,12 +223,11 @@ impl Client {
         if let Some(nodes) = network_nodes.get(most_nodes.0) {
             let pow_feature = String::from("pow");
 
+            // TODO why do we set this in a loop?
```

Comment:
Yeah `network_info.write()` should happen only once

Reply:
I was actually asking why we reset the network info for all nodes

Reply:
Yeah I mean it should only write once and only from one node

Reply:
Just the first node would be fine?

Reply:
Should be